torchrl/objectives/ppo.py

```diff
@@ -756,8 +756,9 @@ def loss_critic(self, tensordict: TensorDictBase) -> torch.Tensor:
                 tgt = target_return.detach()
                 pred = state_value.detach()
                 eps = torch.finfo(tgt.dtype).eps
-                resid = torch.var(tgt - pred, unbiased=False, dim=0)
-                total = torch.var(tgt, unbiased=False, dim=0)
+
+                resid = tree_map(partial(torch.var, unbiased=False, dim=0), tgt - pred)
+                total = tree_map(partial(torch.var, unbiased=False, dim=0), tgt)
```

Comment:
Is this a tree_map over the tensordict leaves?
Not sure we want to keep that in the long-term (it's 50/50 as of today) because it kind of messes up the vmap integration to have TD as a pytree.

I see that `torch.overrides.get_overridable_functions()` has `var` registered so we could simply do that such that `torch.var()` executes `var` over each leaf?